### PR TITLE
chore: tweak log level in production

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Make `stream` and `storage` follow the default level, i.e., `INFO` in production and `DEBUG` for development. `Re`solve #12610.
- Set the level for logs with target `event::` from `TRACE` to `DEBUG`. Considering `release_max_level_debug`, this gives us the possibility to enable them in production without hurting the performance too much (note that the overall max level is `INFO` now, so they can be filtered out in the fast path normally).

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/utils/runtime/src/logger.rs [hunk 1]
```diff
@@ -160,6 +166,7 @@ pub fn init_risingwave_logger(settings: LoggerSettings) {
     // Default filter for logging to stdout and tracing.
     let default_filter = {
         let mut filter = filter::Targets::new()
+            .with_target("foyer", Level::WARN)
```
Comment (xxchan) [inline src/utils/runtime/src/logger.rs:169]: Note for others: `with_target` matches target prefix.

**Review by xxchan — APPROVED** [review]
Changes LGTM

Questions:
- Why "`Re`solve"?
- What do you mean by "the overall max level"?

**Comment by BugenZhao** [comment]
> * What do you mean by "the overall max level"?

There's a [`max_level_hint`](https://docs.rs/tracing/latest/tracing/trait.Subscriber.html#method.max_level_hint). As the verbosity levels of all targets are `<= INFO` in production now, a debug event will be efficiently filtered out according to this hint without the need to lookup the per-target directives.

**Review by MrCroxx — APPROVED** [review]
LGTM

File: src/utils/runtime/src/logger.rs [hunk 2]
```diff
@@ -28,27 +28,33 @@ use tracing_subscriber::{filter, EnvFilter};
...
+            }
+        }
+    }
+}
+
 /// Configure log targets for some `RisingWave` crates.
 ///
-/// Other RisingWave crates will follow the default level (`DEBUG` or `INFO` according to
-/// the `debug_assertions` and `is_ci` flag).
+/// Other RisingWave crates (like `stream` and `storage`) will follow the [`default_level`].
```
[the remark this answers is not in the document]
Comment (BugenZhao) [reply src/utils/runtime/src/logger.rs]: Ok

**Comment by BugenZhao** [comment]
> * Why "`Re`solve"?

Looks like #12610 requires more refactors besides tweaking the log level. For example, eliminate periodical `INFO` events. So let's leave it open for now.

**Pull request merged**